pkg/rabbitmq/cluster.go

```diff
@@ -0,0 +1,284 @@
+	topologyv1 "github.com/openstack-k8s-operators/infra-operator/apis/topology/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/labels"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	rabbitmqv2 "github.com/rabbitmq/cluster-operator/v2/api/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+func ConfigureCluster(
+	cluster *rabbitmqv2.RabbitmqCluster,
+	IPv6Enabled bool,
+	fipsEnabled bool,
+	topology *topologyv1.Topology,
+	nodeselector *map[string]string,
+) *rabbitmqv2.RabbitmqCluster {
+
+	envVars := []corev1.EnvVar{
+		{
+			// The upstream rabbitmq image has /var/log/rabbitmq mode 777, so when
+			// openshift runs the rabbitmq container as a random uid it can still write
+			// the logs there.  The OSP image however has the directory more constrained,
+			// so the random uid cannot write the logs there.  Force it into /var/lib
+			// where it can create the file without crashing.
+			Name:  "RABBITMQ_UPGRADE_LOG",
+			Value: "/var/lib/rabbitmq/rabbitmq_upgrade.log",
+		},
+		{
+			// For some reason HOME needs to be explicitly set here even though the entry
+			// for the random user in /etc/passwd has the correct homedir set.
+			Name:  "HOME",
+			Value: "/var/lib/rabbitmq",
+		},
+		{
+			// The various /usr/sbin/rabbitmq* scripts are really all the same
+			// wrapper shell-script that performs some "sanity checks" and then
+			// invokes the corresponding "real" program in
+			// /usr/lib/rabbitmq/bin.  The main "sanity check" is to ensure that
+			// the user running the command is either root or rabbitmq.  Inside
+			// of an openshift pod, however, the user is neither of these, so
+			// the wrapper script will always fail.
+
+			// By putting the real programs ahead of the wrapper in PATH we can
+			// avoid the unnecessary check and just run things directly as
+			// whatever user the pod has graciously generated for us.
+			Name:  "PATH",
+			Value: "/usr/lib/rabbitmq/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+		},
+	}
+
+	inetFamily := "inet"
+	inetProtocol := "tcp"
+	tlsArgs := ""
+	fipsArgs := ""
+	if IPv6Enabled {
+		inetFamily = "inet6"
+	}
+	erlangInetConfig := fmt.Sprintf("{%s,true}.\n", inetFamily)
+
+	if cluster.Spec.TLS.SecretName != "" {
+		inetProtocol = "tls"
+		tlsArgs = "-ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config"
+		if fipsEnabled {
+			fipsArgs = "-crypto fips_mode true"
+		}
+	}
+
+	envVars = append(envVars, corev1.EnvVar{
+		Name: "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
+		Value: fmt.Sprintf(
+			"-kernel inetrc '/etc/rabbitmq/erl_inetrc' -proto_dist %s_%s %s %s",
+			inetFamily,
+			inetProtocol,
+			tlsArgs,
+			fipsArgs,
+		),
+	}, corev1.EnvVar{
+		Name:  "RABBITMQ_CTL_ERL_ARGS",
+		Value: fmt.Sprintf("-proto_dist %s_%s %s", inetFamily, inetProtocol, tlsArgs),
+	})
+
+	defaultStatefulSet := rabbitmqv2.StatefulSet{
```

Comment:
I'm not sure about that, but do we have a way to pass labels here?

Reply:
Why? Probably can, need to look at the rabbitmqcluster logic

Reply:
ignore this comment, sorry, this goes to the NewRabbitMqCluster function as selectors. I'm resolving now so it doesn't generate confusion